### PR TITLE
chore: replace `ExecutionStage::read_block_with_senders` with `BlockProvider::block_with_senders`

### PR DESCRIPTION
**Pull request by joshieDo**
Moves `read_block_with_senders` from execution stage to `BlockProvider::block_with_senders`.

Also adds: 
`transactions_by_tx_range`
`senders_by_tx_range`

### CODE REVIEW

File: crates/storage/provider/src/providers/database/provider.rs
Comment (onbjerg) [inline crates/storage/provider/src/providers/database/provider.rs:1632]: we need to note here that the tx's do not have valid hashes somehow

**Review by onbjerg — APPROVED** [review]
lgtm modulo note + lint and test failing

File: crates/storage/provider/src/providers/database/provider.rs
Comment (joshieDo) [reply crates/storage/provider/src/providers/database/provider.rs:1638]: added some docs to the implementation on `DatabaseProvider`

**Review by gakonst — COMMENTED** [review]
Can we have 100% testing on each of these methods before merging? This is a great opportunity to generate proper regression tests for all these bindings, which previously bundled a lot of logic.

**Pull request merged**
